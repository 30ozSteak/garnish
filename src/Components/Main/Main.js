import React, { Component } from "react";
import Header from "../../Components/Header/Header";
import Fab from "../../Components/Fab/Fab";
import HomeHeader from "../../Components/HomeHeader/HomeHeader";
import Featured from "../../Components/Featured/Featured";
import NewsCard from "../../Components/NewsCard/NewsCard";
import { fetchNews } from "../../Helpers/fetcher";
import { updateNews } from "../../Actions/index";
import { connect } from "react-redux";

export class Main extends Component {
  handleNewsLink = async type => {
    const news = await fetchNews(type);
    this.props.updateNews(news.articles);
  };

  render() {
    return (
      <div>
        <Header />
        <Fab />
        <HomeHeader />
        <Featured handleNewsLink={this.handleNewsLink} />
        <div className="biggest-news-box">
          <NewsCard news={this.props.news} />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = ({ news }) => ({
  news
});

export const mapDispatchToProps = dispatch => ({
  updateNews: news => dispatch(updateNews(news))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Main);
