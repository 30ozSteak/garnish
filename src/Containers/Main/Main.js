import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchNews } from "../../Helpers/fetcher";
import { updateNews } from "../../Actions/index";

import Fab from "../../Components/Fab/Fab";
import Featured from "../../Components/Featured/Featured";
import Header from "../../Containers/Header/Header";
import HomeHeader from "../../Components/HomeHeader/HomeHeader";
import NewsContainer from "../../Containers/NewsContainer/NewsContainer";

export class Main extends Component {
  handleNewsLink = async type => {
    const news = await fetchNews(type);
    this.props.updateNews(news.articles);
  };

  render() {
    return (
      <div className="busted-box">
        <Header />
        <Fab />
        <HomeHeader />
        <Featured handleNewsLink={this.handleNewsLink} />
        <div className="biggest-news-box">
          <NewsContainer />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = ({ news }) => ({
  news
});

export const mapDispatchToProps = dispatch => ({
  updateNews: news => dispatch(updateNews(news))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Main);
