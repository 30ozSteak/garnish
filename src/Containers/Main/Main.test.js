import React from "react";
import { shallow } from "enzyme";
import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps } from "./Main";

import { Main } from "./Main";

describe("Main", () => {
  it("should resemble the snapshot", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
