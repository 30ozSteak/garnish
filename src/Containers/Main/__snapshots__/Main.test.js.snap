// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should resemble the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Menu />,
        <HomeHeader />,
        <Featured
          handleNewsLink={[Function]}
        />,
        <div
          className="biggest-news-box"
        >
          <Connect(NewsContainer) />
        </div>,
      ],
      "className": "busted-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleNewsLink": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(NewsContainer) />,
          "className": "biggest-news-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Menu />,
          <HomeHeader />,
          <Featured
            handleNewsLink={[Function]}
          />,
          <div
            className="biggest-news-box"
          >
            <Connect(NewsContainer) />
          </div>,
        ],
        "className": "busted-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleNewsLink": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(NewsContainer) />,
            "className": "biggest-news-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
